feat: 채팅 미션완료 퍼널 추가

Changsoon-Yun opened the pull request:
### 작업내역

1. 채팅창에서 미션완료 제출 팝업 위치 수정
2. 미션완료 퍼널 추가
3. 미션완료 exit 모달 추가

<img src="https://github.com/user-attachments/assets/19dbc140-068f-4cdd-a611-bf5acfc19c71" width="200" height="450" />
<img src="https://github.com/user-attachments/assets/473d8a74-9a9c-4aa2-ae7a-092bc027a063" width="200" height="450" />
<img src="https://github.com/user-attachments/assets/544a629b-ab33-4b24-b454-83176f080e2b" width="200" height="450" />
<img src="https://github.com/user-attachments/assets/58387f6e-0eec-4a90-84de-044f4e8af7b0" width="200" height="450" />

Changsoon-Yun committed:
feat: submission page 초안 추가

Changsoon-Yun committed:
feat: 체크박스 추가

Changsoon-Yun committed:
feat: 미션완료 제출 퍼널 추가

Changsoon-Yun committed:
fix: 함수 export 형식 수정

dusunax commented on src/app/(pages)/chat/[id]/_components/ChatSendFooter.tsx:
```diff
@@ -1,29 +1,38 @@
   const labelRef = useRef<HTMLLabelElement>(null);
+  const popupRef = useRef<HTMLDivElement>(null);
+
+  useClickAway(popupRef, () => {
+    setIsOpen(false);
+  });
```
MissonPopup에 핸들러 함수를 넘기지 않고 ref와 보조함수를 활용한 점이 좋네요.
팝업 내부에서 함수를 알 필요가 없고, 클릭/터치 이벤트 구독만 하니까 깔끔한 구조가 된 것 같습니다

dusunax commented on src/app/(pages)/chat/[id]/_components/OtherChat.tsx:
깔끔👍

dusunax commented on src/app/(pages)/chat/[id]/submission/_components/SubmissionCheckList.tsx:
```diff
@@ -0,0 +1,60 @@
+  };
+
+  return (
+    <div>
```
padding bottom도 있는 FunnelUi 래퍼 많이 활용해주세여🙏

dusunax commented on src/app/(pages)/chat/[id]/submission/_components/SubmissionDescription.tsx:
```diff
@@ -0,0 +1,51 @@
+        </TextArea>
+      </FieldWrapper>
+      <ButtonWrapper>
+        <Button
+          onClick={() => router.push(`/chat/${id}`)}
+          disabled={!description || (pictures !== null && pictures.length > MAX_PICTURES)}
+        >
+          제출하기
+        </Button>
```
**네비게이션:**
제출하기를 완료한 후, 채팅방(/chat/id)의 헤더의 뒤로 가기 버튼을 누르면 미션 제출 페이지(/chat/123/submission)로 돌아옵니다!
/chat/id에서는 /chat으로 이동해야 할 것 같아요

dusunax reviewed (commented):
연휴 끝나고 출근하느라 힘드셨을텐데 고생하셨습니다

몇가지 코멘트 남겼습니다😄
그리고 전체적인 여백 사이즈 한 번 체크 부탁드립니다~!

dusunax commented on src/app/(pages)/chat/[id]/submission/_components/SubmissionLeaveModal.tsx:
```diff
@@ -0,0 +1,57 @@
+        <div className='mx-6 flex-1' ref={modalRef}>
+          <div className='flex flex-col break-keep rounded-[20px] bg-white px-6 py-4'>
+            <div className='mb-8 mt-6 flex flex-col gap-1 text-center'>
+              <div className='text-xl font-medium text-v1-text-primary-700'>미션 등록을 그만 두시겠어요?</div>
```
+) 이거 문구가 미션 등록이 아닐 것 같은데
회의 때 체크해봐야할 것 같습니다🤔

Changsoon-Yun replied:
그러게요 등록이아니라 제출인데 ㅋㅋㅋ제가 그냥 수정할게요 제출로

Changsoon-Yun committed:
Merge branch 'dev' into feat/chat-submission

Changsoon-Yun committed:
fix: pr 수정커밋
1. padding등 여백 수정
2. ChatRoomPage.tsx router 훅 추가
3. ChatPage 임시 주석처리
4. ModalProvider.tsx 병합충돌에러 수정
5. FunnelUi 추가
6. 텍스트 수정

dusunax reviewed (approved):
고생하셨습니다!
